public/gatk-utils/src/main/java/org/broadinstitute/gatk/utils/MannWhitneyU.java

```diff
@@ -154,18 +163,18 @@ public void add(Number n, USet set) {
      */
     @Requires({"m > 0","n > 0"})
     @Ensures({"result != null", "! Double.isInfinite(result.getFirst())", "! Double.isInfinite(result.getSecond())"})
-    protected static Pair<Double,Double> calculateP(int n, int m, long u, boolean twoSided, ExactMode exactMode) {
+    protected static Pair<Double,Double> calculateP(int n, int m, double u, boolean twoSided, Map<Integer, Integer> tieStructure, ExactMode exactMode) {
         Pair<Double,Double> zandP;
         if ( n > 8 && m > 8 ) {
             // large m and n - normal approx
-            zandP = calculatePNormalApproximation(n,m,u, twoSided);
+            zandP = calculatePNormalApproximation(n,m,u,tieStructure, twoSided);
         } else if ( n > 5 && m > 7 ) {
```

Comment:
Huh.  This is...weird.  The bounding regions are weird and it's not actually using the uniform approximation.  Why not just clean up the if cases and add a TODO comment in that the uniform approximation is worth looking into?